doc/rpc/pathfinder_ws.json

```diff
@@ -0,0 +1,181 @@
+    "info": {
+        "title": "Pathfinder websocket RPC API",
+        "version": "0.1",
+        "description": "Provides additional (pathfinder specific) methods over and above the Starknet RPC API"
+    },
+    "methods": [
+        {
+            "name": "pathfinder_subscribe",
+            "summary": "Open a new websocket subscription",
+            "description": "Creates a websocket stream which will fire events of the requested subscription type. Only available for websocket connections.",
+            "params": [
+                {
+                    "name": "kind",
+                    "summary": "The type of subscription",
+                    "required": true,
+                    "schema": {
+                        "type": "string",
+                        "enum": [
+                            "newHeads"
+                        ]
+                    }
+                }
+            ],
+            "result": {
+                "name": "subscription ID",
+                "description": "An identifier for this subscription stream used to associate events with this subscription.",
+                "schema": {
+                    "type": "integer"
+                }
+            }
+        },
+        {
+            "name": "pathfinder_unsubscribe",
+            "summary": "Closes a websocket subscription",
+            "description": "Creates a websocket stream which will fire events of the requested subscription type. Only available for websocket connections.",
+            "params": [
+                {
+                    "name": "subscription ID",
+                    "summary": "The subscription to close",
+                    "required": true,
+                    "schema": {
+                        "type": "integer"
+                    }
+                }
+            ],
+            "result": {
+                "name": "Unsubscription result",
+                "description": "True if the unsubscription was successful",
+                "schema": {
+                    "type": "boolean"
+                }
+            }
+        },
+        {
+            "name": "pathfinder_subscription",
+            "summary": "A subscription event notification sent by the node.",
+            "params": [],
+            "result": {
+                "name": "Subscription event",
+                "schema": {
+                    "subscription": {
+                        "name": "Subscription ID",
+                        "summary": "The subscription this event is for",
+                        "required": true,
+                        "type": "integer"
+                    },
+                    "result": {
+                        "summary": "The event data",
+                        "required": true,
+                        "schema": {
+                            "summary": "Sent by 'newHeads' subscriptions",
+                            "$ref": "#/components/schemas/BLOCK_HEADER"
+                        }
```

Comment:
I think this must be a oneof since we will one day hopefully add more event types? Or does that not matter?

Reply:
Whether we keep the `oneOf` or not, adding another event type would be a breaking change anyway.
We could also choose to differentiate the subscription types using the method name. That's where the OpenRPC spec doesn't fit really well for subscriptions :D I looked for the Ethereum OpenRPC spec for websocket and couldn't find it. Maybe that's one of the reasons why.

On the topic of anticipating future changes, I'm confident that this is why the nested result exists: to be able to pass errors that would not close the subscription. We currently have none of these.

Reply:
The caller should be able to map the event type from the subscription ID since he knows what subscription type it is (so no need for method name), so I thought indicating that this is an "open" enum so long is a good start for anyone trying to decode this so they don't have to make changes again.

Reply:
> On the topic of anticipating future changes, I'm confident that this is why the nested result exists: to be able to pass errors that would not close the subscription. We currently have none of these.

Would that not simply be another output type? So `oneof` of `BLOCK_HEADER` + `ERROR_XX` etc.

Reply:
I guess it depends how flat or nested you want to be. Should a normal error that continues operating really be an error? Or just another output value?

Reply:
I think it's makes slightly more sense to have subscription-breaking errors in the error section.

Now on the questions of the nested result and the `oneOf`, how about we don't anticipate and remove both? Whatever we add in the future, it will be a breaking change, and the bigger the change in the spec, the more explicit the breaking change will be.

Reply:
Sounds good to me 👍

Reply:
Alright, I'm on it.